draft-openid-ipsie-sl1-profile.md

```diff
@@ -183,17 +181,17 @@ Note 3: This claim is not currently defined in OpenID Connect, and should be pul
 
 For the authorization code flow, OpenID Providers:
 
-* shall require the value of `response_type` described in [RFC6749] to be `code`;
-* shall require PKCE [RFC7636] with S256 as the code challenge method (see Note 1 below);
-* shall require an exact match of a registered redirect URI as described in {{Section 2.1 of RFC9700}};
-* shall issue authorization codes with a maximum lifetime of 60 seconds;
-* shall support "Authorization Code Binding to DPoP Key" (as required by {{Section 10.1 of RFC9449}});
-* shall return an iss parameter in the authorization response according to [RFC9207];
-* shall not transmit authorization responses over unencrypted network connections, and, to this end, shall not allow redirect URIs that use the "http" scheme;
-* shall reject an authorization code (Section 1.3.1 of [RFC6749]) if it has been previously used;
-* shall not use the HTTP 307 status code when redirecting a request that contains user credentials to avoid forwarding the credentials to a third party accidentally (see {{Section 4.12 of RFC9700}});
-* should use the HTTP 303 status code when redirecting the user agent using status codes;
-* shall support `nonce` parameter values up to 64 characters in length, may reject `nonce` values longer than 64 characters.
+* MUST require the value of `response_type` described in [RFC6749] to be `code`;
+* MUST require PKCE [RFC7636] with S256 as the code challenge method (see Note 1 below);
+* MUST require an exact match of a registered redirect URI as described in {{Section 2.1 of RFC9700}};
+* MUST issue authorization codes with a maximum lifetime of 60 seconds;
+* MUST support "Authorization Code Binding to DPoP Key" (as required by {{Section 10.1 of RFC9449}});
+* MUST return an `iss` parameter in the authorization response according to [RFC9207];
+* MUST NOT transmit authorization responses over unencrypted network connections, and, to this end, MUST NOT allow redirect URIs that use the `http` scheme;
+* MUST reject an authorization code (Section 1.3.1 of [RFC6749]) if it has been previously used;
+* MUST NOT use the HTTP 307 status code when redirecting a request that contains user credentials to avoid forwarding the credentials to a third party accidentally (see {{Section 4.12 of RFC9700}});
+* SHOULD use the HTTP 303 status code when redirecting the user agent using status codes;
+* MUST support `nonce` parameter values up to 64 characters in length, and MAY reject `nonce` values longer than 64 characters.
```

Comment:
the MAY clause could create interop issues.  I won't reject this due to this issue, rather, we should work to clarify this, perhaps by stating a min/max length.

Reply:
Yeah let's discuss this in a separate issue since this PR doesn't change this requirement